### PR TITLE
auto-increment: update doc for the new implementation

### PR DESCRIPTION
**Pull request by tiancaiamao**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
https://github.com/pingcap/tidb/pull/38449

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: auto-increment.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ aliases: ['/docs-cn/dev/auto-increment/']
 
 出于性能原因，自增编号是系统批量分配给每台 TiDB 服务器的值（默认 3 万个值），因此自增编号能保证唯一性，但分配给 `INSERT` 语句的值仅在单台 TiDB 服务器上具有单调性。
 
+> **注意：**
+>
+> 如果要求自增编号单调，并且 TiDB 版本在 v6.4.0 及以上，推荐使用 [MySQL 兼容模式](/auto-increment.md#MySQL-兼容模式)
```
Comment (Oreoxmt) [inline auto-increment.md:23]:
```suggestion
> 如果要求自增编号单调，并且 TiDB 版本在 v6.4.0 及以上，推荐使用 [MySQL 兼容模式](/auto-increment.md#mysql-兼容模式)。
```

File: auto-increment.md [hunk 2]
```diff
@@ -322,6 +326,27 @@ SELECT * FROM t;
 
 从 v3.0.9 和 v4.0.rc-1 开始，和 MySQL 的行为类似，自增列隐式分配的值遵循 session 变量 `@@auto_increment_increment` 和 `@@auto_increment_offset` 的控制，其中自增列隐式分配的值 (ID) 将满足式子 `(ID - auto_increment_offset) % auto_increment_increment == 0`。
 
+## MySQL 兼容模式
+
+从 v6.4.0 开始，TiDB 实现了中心化分配的服务，可以支持 TiDB 实例不缓存数据，而是每次请求都访问中心化服务获取 ID。
+
+当前中心化分配服务内置在 TiDB 进程，类似于 DDL Owner 的工作模式。有一个 TiDB 实例将充当 "主" 的角色提供 ID 分配服务，而其它的 TiDB 实例将充当 "备" 角色。当 "主" 节点发生故障时，会自动进行 "主备切换"，从而保证中心化服务的高可用。
+
+MySQL 兼容模式的使用方式是，建表时将 `AUTO_ID_CACHE` 设置为 1：
```
Comment (Oreoxmt) [inline auto-increment.md:335]:
```suggestion
当前中心化分配服务内置在 TiDB 进程，类似于 DDL Owner 的工作模式。有一个 TiDB 实例将充当“主”的角色提供 ID 分配服务，而其它的 TiDB 实例将充当“备”角色。当“主”节点发生故障时，会自动进行“主备切换"，从而保证中心化服务的高可用。

MySQL 兼容模式的使用方式是，建表时将 `AUTO_ID_CACHE` 设置为 `1`：
```

File: auto-increment.md [hunk 3]
```diff
@@ -322,6 +326,27 @@ SELECT * FROM t;
...
+```
+
+> **注意：**
+>
+> 无论新版本或旧版本，`AUTO_ID_CACHE` 设置为 1 都表明 TiDB 不再缓存 ID，但是具体对应的实现方式不一样。
+> 早期的版本设置 `AUTO_ID_CACHE` 为 1 之后性能非常差，因为每次分配 ID 都需要通过一个 TiKV 事务完成 `AUTO_INCREMENT` 值的持久化修改。
+> v6.4.0 以上版本(包含 v6.4.0)，由于引入了中心化的分配服务，`AUTO_INCREMENT` 值的修改只是在服务进程中的一个内存操作，更加轻量。
+
+使用 MySQL 兼容模式后，能保证 ID **唯一**，**单调递增**，行为几乎跟 MySQL 完成一致，即使跨 TiDB 实例访问，ID 也不会出现回退。
+仅当中心化服务的 "主" 异常崩溃时，有可能会造成少量 ID 不连续，出现空洞。这是因为主备切换时，"备" 需要丢弃一部分之前的 "主" 可能分配出去的 ID，以保证 ID 不出现重复。
```
Comment (Oreoxmt) [inline auto-increment.md:348]:
```suggestion

> **注意：**
>
> 在 TiDB 各个版本中，`AUTO_ID_CACHE` 设置为 `1` 都表明 TiDB 不再缓存 ID，但是不同版本的实现方式不一样：
>
> - 对于 TiDB v6.4.0 之前的版本，由于每次分配 ID 都需要通过一个 TiKV 事务完成 `AUTO_INCREMENT` 值的持久化修改，因此设置 `AUTO_ID_CACHE` 为 `1` 会出现性能下降。
> - 对于 v6.4.0 及以上版本，由于引入了中心化的分配服务，`AUTO_INCREMENT` 值的修改只是在服务进程中的一个内存操作，更加轻量。

使用 MySQL 兼容模式后，能保证 ID **唯一**、**单调递增**，行为几乎跟 MySQL 完成一致。即使跨 TiDB 实例访问，ID 也不会出现回退。只有当中心化服务的“主”异常崩溃时，才有可能造成少量 ID 不连续，出现空洞。这是因为主备切换时，“备”需要丢弃一部分之前的“主“可能分配出去的 ID，以保证 ID 不出现重复。
```

**Comment by Oreoxmt** [comment]
DNM until https://github.com/pingcap/tidb/pull/38449 is merged

File: auto-increment.md [hunk 4]
```diff
@@ -322,6 +326,27 @@ SELECT * FROM t;
...
+MySQL 兼容模式的使用方式是，建表时将 `AUTO_ID_CACHE` 设置为 1：
+
+```sql
+CREATE TABLE t(a int AUTO_INCREMENT key) AUTO_ID_CACHE 1;
+```
+
+> **注意：**
+>
+> 无论新版本或旧版本，`AUTO_ID_CACHE` 设置为 1 都表明 TiDB 不再缓存 ID，但是具体对应的实现方式不一样。
+> 早期的版本设置 `AUTO_ID_CACHE` 为 1 之后性能非常差，因为每次分配 ID 都需要通过一个 TiKV 事务完成 `AUTO_INCREMENT` 值的持久化修改。
```
Comment (lcwangchao) [inline auto-increment.md:344]: "性能非常差" 是不是可以改个说法，比如相比新版本性能相对较差之类的

**Review by lcwangchao — APPROVED** [review]

File: auto-increment.md (hunk 4 above)
Comment (Oreoxmt) [reply auto-increment.md:344]: 如果有测试数据可以加一下

Comment (tiancaiamao) [reply auto-increment.md]: 当下测试还没有搞完

File: auto-increment.md [hunk 5]
```diff
@@ -18,6 +18,10 @@ aliases: ['/docs-cn/dev/auto-increment/']
 
 出于性能原因，自增编号是系统批量分配给每台 TiDB 服务器的值（默认 3 万个值），因此自增编号能保证唯一性，但分配给 `INSERT` 语句的值仅在单台 TiDB 服务器上具有单调性。
 
+> **注意：**
+>
+> 如果要求自增编号单调，并且 TiDB 版本在 v6.4.0 及以上，推荐使用 [MySQL 兼容模式](/auto-increment.md#mysql-兼容模式)。
```
Comment (Oreoxmt) [inline auto-increment.md:23]:
```suggestion
> 如果要求自增编号在所有 TiDB 实例上具有单调性，并且你的 TiDB 版本在 v6.4.0 及以上，推荐使用 [MySQL 兼容模式](/auto-increment.md#mysql-兼容模式)。
```

File: auto-increment.md [hunk 6]
```diff
@@ -322,6 +326,28 @@ SELECT * FROM t;
 
 从 v3.0.9 和 v4.0.rc-1 开始，和 MySQL 的行为类似，自增列隐式分配的值遵循 session 变量 `@@auto_increment_increment` 和 `@@auto_increment_offset` 的控制，其中自增列隐式分配的值 (ID) 将满足式子 `(ID - auto_increment_offset) % auto_increment_increment == 0`。
 
+## MySQL 兼容模式
+
+从 v6.4.0 开始，TiDB 实现了中心化分配的服务，可以支持 TiDB 实例不缓存数据，而是每次请求都访问中心化服务获取 ID。
```
Comment (Oreoxmt) [inline auto-increment.md:331]: [实现原理](https://docs.pingcap.com/zh/tidb/dev/auto-increment#%E5%AE%9E%E7%8E%B0%E5%8E%9F%E7%90%86) 部分是不是需要更新为是从中心化服务获取 ID 而不是缓存呢

Comment (tiancaiamao) [reply auto-increment.md:331]: 实现原理先不更新这块的内容吧，因为默认情况下使用的都是 AUTOID_CACHE != 1 的
我们得按默认实现解释

单调性保证也是针对默认设置下的。而且文档最开头已经写过如果有要求，建议使用 MySQL 兼容模式的描述了

File: auto-increment.md [hunk 7]
```diff
@@ -322,6 +326,27 @@ SELECT * FROM t;
...
+```
+
+> **注意：**
+>
+> 在 TiDB 各个版本中，`AUTO_ID_CACHE` 设置为 `1` 都表明 TiDB 不再缓存 ID，但是不同版本的实现方式不一样：
+>
+> - 对于 TiDB v6.4.0 之前的版本，由于每次分配 ID 都需要通过一个 TiKV 事务完成 `AUTO_INCREMENT` 值的持久化修改，因此设置 `AUTO_ID_CACHE` 为 `1` 会出现性能下降。
+> - 对于 v6.4.0 及以上版本，由于引入了中心化的分配服务，`AUTO_INCREMENT` 值的修改只是在 TiDB 服务进程中的一个内存操作，相较于之前版本更加轻量。
+
+使用 MySQL 兼容模式后，能保证 ID **唯一**、**单调递增**，行为几乎跟 MySQL 完全一致。即使跨 TiDB 实例访问，ID 也不会出现回退。只有当中心化服务的“主” TiDB 实例异常崩溃时，才有可能造成少量 ID 不连续，出现空洞。这是因为主备切换时，“备”需要丢弃一部分之前的“主”可能分配出去的 ID，以保证 ID 不出现重复。
```
Comment (shichun-0415) [inline auto-increment.md:348]:
```suggestion
使用 MySQL 兼容模式后，能保证 ID **唯一**、**单调递增**，行为几乎跟 MySQL 完全一致。即使跨 TiDB 实例访问，ID 也不会出现回退。只有当中心化服务的“主” TiDB 实例异常崩溃时，才有可能造成少量 ID 不连续。这是因为主备切换时，“备” 节点需要丢弃一部分之前的“主” 节点可能已经分配的 ID，以保证 ID 不出现重复。
```

File: auto-increment.md [hunk 8]
```diff
@@ -322,6 +326,27 @@ SELECT * FROM t;
...
+```sql
+CREATE TABLE t(a int AUTO_INCREMENT key) AUTO_ID_CACHE 1;
+```
+
+> **注意：**
+>
+> 在 TiDB 各个版本中，`AUTO_ID_CACHE` 设置为 `1` 都表明 TiDB 不再缓存 ID，但是不同版本的实现方式不一样：
+>
+> - 对于 TiDB v6.4.0 之前的版本，由于每次分配 ID 都需要通过一个 TiKV 事务完成 `AUTO_INCREMENT` 值的持久化修改，因此设置 `AUTO_ID_CACHE` 为 `1` 会出现性能下降。
+> - 对于 v6.4.0 及以上版本，由于引入了中心化的分配服务，`AUTO_INCREMENT` 值的修改只是在 TiDB 服务进程中的一个内存操作，相较于之前版本更加轻量。
```
Comment (shichun-0415) [inline auto-increment.md:346]: 操作`更加轻量`，这个描述可能有点抽象。能否修改为
> `AUTO_INCREMENT` 值的修改只是在 TiDB 服务进程中的一个内存操作，消耗的时间将大大缩短（预期 0.5ms 以内）。

**Review by shichun-0415 — APPROVED** [review]
Rest LGTM

File: auto-increment.md (hunk 8 above)
Comment (tiancaiamao) [reply auto-increment.md:346]: (预期 0.5 ms 以内)  这个受网络，跟当前的负载等因素影响太多了，直接放数据不太好

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: auto-increment.md [hunk 9]
```diff
@@ -322,6 +326,27 @@ SELECT * FROM t;
 
 从 v3.0.9 和 v4.0.rc-1 开始，和 MySQL 的行为类似，自增列隐式分配的值遵循 session 变量 `@@auto_increment_increment` 和 `@@auto_increment_offset` 的控制，其中自增列隐式分配的值 (ID) 将满足式子 `(ID - auto_increment_offset) % auto_increment_increment == 0`。
 
+## MySQL 兼容模式
+
+从 v6.4.0 开始，TiDB 实现了中心化分配自增 ID 的服务，可以支持 TiDB 实例不缓存数据，而是每次请求都访问中心化服务获取 ID。
```
Comment (Oreoxmt) [inline auto-increment.md:331]:
```suggestion
从 v6.4.0 开始，TiDB 实现了中心化分配自增 ID 的服务，可以支持 TiDB 实例不缓存数据，而是每次请求都访问中心化服务获取 ID。

> **警告：**
>
> 当前该功能为实验特性，不建议在生产环境中使用。
```

File: auto-increment.md [hunk 10]
```diff
@@ -18,6 +18,10 @@ aliases: ['/docs-cn/dev/auto-increment/']
 
 出于性能原因，自增编号是系统批量分配给每台 TiDB 服务器的值（默认 3 万个值），因此自增编号能保证唯一性，但分配给 `INSERT` 语句的值仅在单台 TiDB 服务器上具有单调性。
 
+> **注意：**
+>
+> 如果要求自增编号在所有 TiDB 实例上具有单调性，并且你的 TiDB 版本在 v6.4.0 及以上，推荐使用 [MySQL 兼容模式](#mysql-兼容模式)。
```
Comment (Oreoxmt) [inline auto-increment.md:23]:
```suggestion
> 如果要求自增编号在所有 TiDB 实例上具有单调性，并且你的 TiDB 版本在 v6.4.0 及以上，推荐使用 [MySQL 兼容模式](#mysql-兼容模式)（实验特性）。
```

File: mysql-compatibility.md [hunk 11]
```diff
@@ -49,7 +49,7 @@ aliases: ['/docs-cn/dev/mysql-compatibility/','/docs-cn/dev/reference/mysql-comp
 
 ### 自增 ID
 
-- TiDB 的自增列既能保证唯一，也能保证在单个 TiDB server 中自增，但不保证多个 TiDB server 中自增，不保证自动分配的值的连续性。不建议将缺省值和自定义值混用，若混用可能会收到 `Duplicated Error` 的错误信息。
+- TiDB 的自增列既能保证唯一，也能保证在单个 TiDB server 中自增，使用 [`AUTO_INCREMENT` MySQL 兼容模式](/auto-increment.md#mysql-兼容模式)能保证多个 TiDB server 中自增，不保证自动分配的值的连续性。不建议将缺省值和自定义值混用，若混用可能会收到 `Duplicated Error` 的错误信息。
```
Comment (shichun-0415) [inline mysql-compatibility.md:52]:
```suggestion
- TiDB 的自增列既能保证唯一，也能保证在单个 TiDB server 中自增，使用 [`AUTO_INCREMENT` MySQL 兼容模式](/auto-increment.md#mysql-兼容模式)能保证多个 TiDB server 中自增 ID，但不保证自动分配的值的连续性。不建议将缺省值和自定义值混用，若混用可能会收到 `Duplicated Error` 的错误信息。
```

**Pull request merged**